roofit/roofitcore/test/testRooFuncWrapper.cxx

```diff
@@ -488,8 +493,120 @@ FactoryTestParams param6{"GaussianExtended",
                          },
                          1e-4,
                          /*randomizeParameters=*/false};
+namespace {
+void getDataHistModel(RooWorkspace &ws)
+{
+   RooRealVar x("x", "x", 6, 0, 20);
+   RooPolynomial p("p", "p", x, RooArgList(0.01, -0.01, 0.0004));
+
+   // Sample 500 events from p
+   x.setBins(10);
+   std::unique_ptr<RooDataSet> data1{p.generate(x, 500)};
+
+   // Create a binned dataset with 10 bins and 500 events
+   std::unique_ptr<RooDataHist> hist1{p.generateBinned(x, 500)};
+
+   // Represent data in dh as pdf in x
+   RooHistPdf histpdf("histpdf", "histpdf", x, *hist1, 0);
+
+   RooRealVar mean("mean", "mean of gaussian", 6, 5, 10);
+   RooRealVar sigma("sigma", "width of gaussian", 1.0, .01, 3.0);
+
+   RooGaussian gauss("gauss", "gauss", x, mean, sigma);
+   RooRealVar frac("frac", "faction of histpdf", 0.5, 0, 1);
+   RooAddPdf model("model", "model", {histpdf, gauss}, frac);
+
+   ws.import(model);
+   ws.defineSet("observables", {x});
+}
+} // namespace
+
+/// Test based on rf706 tutorial
+FactoryTestParams param7{"HistPdf", getDataHistModel,
+                         [](RooAbsPdf &pdf, RooAbsData &data, RooWorkspace &) {
+                            return std::unique_ptr<RooAbsReal>{pdf.createNLL(data, RooFit::BatchMode("cpu"))};
+                         },
+                         1e-4,
+                         /*randomizeParameters=*/true};
+
+namespace {
+void getDataHistFuncModel(RooWorkspace &ws)
+{
+   using namespace RooFit;
+
+   int nEvents = 1000;
+   int numBins = 5;
+
+   ws.factory("x[0, 0, " + std::to_string(3 * numBins) + "]");
+
+   auto &x = *ws.var("x");
+   x.setBins(numBins);
+
+   // The parameters represent the expected events in each bin
+   RooArgList params = ParamHistFunc::createParamSet(ws, "gamma", x, 0, 1000);
+   for (auto *param : static_range_cast<RooRealVar *>(params)) {
+      param->setVal(nEvents / numBins);
+   }
+   ParamHistFunc phf{"phf", "phf", x, params};
+   RooRealSumPdf pdf{"pdf", "pdf", phf, RooArgList{1.0}};
```

Comment:
```suggestion
   // Constructing a bin width function is a bit tedious. But including it is
   // required to use the binned likelihood optimization.
   RooDataHist dataHist{"data_hist", "data_hist", x};
   RooHistFunc histFunc{"hist_func", "hist_func", x, dataHist};
   RooBinWidthFunction bwf{"bwf", "bwf", histFunc, true};

   ParamHistFunc phf{"phf", "phf", x, params};
   RooProduct prod{"prod", "prod", phf, bwf};
   RooRealSumPdf pdf{"pdf", "pdf", prod, RooArgList{1.0}};
```
For the binned likelihood optimiazation to work after #13024, we need to include a RooBinWidthFunction.